compiler-core/src/parse.rs

```diff
@@ -401,9 +401,9 @@ where
                 let _ = self.next_tok();
                 let mut message = None;
                 if self.maybe_one(&Token::As).is_some() {
-                    let (_, l, e) = self.expect_string()?;
-                    message = Some(l);
-                    end = e;
+                    let msg_expr = self.expect_expression()?;
```

Comment:
Parsing an expression unit will do the trick here for the parsing bug

Reply:
Wouldn't parsing an expression unit in this context disable string concatenation ? 
eg: ```panic as "Hello" <> " There"  ``` or  ``` panic as var1 <> var2 ```

Reply:
It would be written like this:

```gleam
panic as { "Hello " <> name }

Reply:
Oh yeah ! Thank you.